### ISSUE TITLE
Benchmarking is not imported from configuration file

### ISSUE THREAD

**Issue by yasintha91** [Helmholtz-AI-Energy/perun#172]
I am using a custom configuration file for measurements. Benchmarking units are not given as I expected. I am executing the command from the folder where the python script is and giving the configuration file which is in the same folder. All the other custom inputs are changed as expected, only noticing this issue for benchmarking units. I want to change the time units as it ranges around milliseconds.

perun -c .perun.ini monitor [script]

configuration file

[post-processing]
power_overhead = 0
pue = 1.0
emissions_factor = 417.8
price_factor = 0.3251
price_unit = €
[monitor]
sampling_period = 1
include_backends =
include_sensors =
exclude_backends =
exclude_sensors =
[output]
app_name
run_id
format = text
data_out = ./perun_results
[benchmarking]
rounds = 1
warmup_rounds = 0
metrics = runtime,energy
region_metrics = runtime,power,energy
[benchmarking.units]
joule = k
second = m
percent =
watt =
byte = G
[debug]
log_lvl = WARNING

**Comment by JuanPedroGHM** [comment]
Hi, 

The benchmarking units in the configuration change the output when using the option ```--format bench```. This will replace the text file for a json file which will use the units that you are using. 

What is also relevant is the length of your script or your functions. The sampling period that you are using 1, meaning you are getting values once every second. If your functions or program are on the millisecond range, you might one to change the sampling period to 0.001. 

If you give me a bit more information on what you are trying to benchmark and an example script, I can help you with the perun setup.

**Comment by yasintha91** [comment]
It is about measuring model inference energy consumption. It is around millisecond range if we consider one sample in the function. But using sampling period 0.001 is not possible I guess. I get below warning message. So right now I am using batch-wise calculating and normalizing it for one sample.

28-04-2025 03:09:17 : WARNING : configuration : sanitize_config -- Invalid sampling period 0.001. Should be a number higher than 0.1 . Defaulting to 1.


Part of the script: model inference.


@monitor()
def model_inference_4(model, inputs, p=None):
    return model(inputs, sampler=p)
    
@monitor()
def model_inference_3(model, inputs, p=None):
    return model(inputs, sampler=p)
    
@monitor()
def model_inference_2(model, inputs, p=None):
    return model(inputs, sampler=p)

@monitor()
def model_inference_1(model, inputs, p=None):
    return model(inputs, sampler=p)
    
@monitor()
def model_inference_5(model, inputs, p=None):
    return model(inputs, sampler=p)



test_step_functions = {
    0.2: model_inference_1,
    0.4: model_inference_2,
    0.6: model_inference_3,
    0.8: model_inference_4,
    1.0: model_inference_5
}
   
   
def test_step(model, dataset, device,p=1.0):
    model.eval()
    correct = torch.tensor(0.0, device = device)
    with torch.no_grad():
        for batch in dataset:
            inputs, targets = batch[0].to(device), batch[1].to(device)
            
            outputs = test_step_functions[p](model,inputs,p=p)
            correct += torch.sum(outputs.argmax(dim = 1) == targets).float()
            #break
    acc = 100. * correct / len(dataset.dataset)
    print("Accuracy : {:.0f}%".format(acc.item()))
    return acc.item()

**Comment by yasintha91** [comment]
1. And one thing is if I use **--format bench**, it does not give the standard deviation for monitored function as well. That is one thing I am interested in. 
2. Eventhough, I defined energy in region_metrics, energy results are not given for monitored functions. But of course, I am able to get them using runtime and power. But it would be easy to get it straight from the results.

**Comment by JuanPedroGHM** [comment]
You are right, I forgot the limit is set at 0,1. That is because most of the energy measurement interfaces have a slow update rate. Measuring with larger batches is a good solution.

Regarding the bench format, the json objects should have a ```"range"``` attribute in most of the entries, as long as each of the functions is run more than two times. It is called range due to an old benchmarking dashboard that I was using, but the value that is written is the standard deviation. 

It also only supports power and not energy at the moment, as shown in the [docs](https://perun.readthedocs.io/en/latest/configuration.html). This is again a leftover of my previous benchmarking dashboard. 

I'm slowly working towards expanding the data storage and output formats to make them more configurable and accessible, but it might take a while until it is fully done. Rough plans are here (#158)  and in other issues.

**Comment by yasintha91** [comment]
Thanks. I think I can work with current repository.
